cli/npm/managed/resolvers/local.rs

```diff
@@ -472,6 +493,74 @@ async fn sync_resolution_with_fs(
     }
   }
 
+
+  // 6. Set up `node_modules/.bin` entries for packages that need it.
+  for (package, package_path) in &*bin_entries_to_setup.lock() {
+    let package = snapshot.package_from_id(&package.id).unwrap();
+    if let Some(bin_entries) = &package.bin {
+      match bin_entries {
+        deno_npm::registry::NpmPackageVersionBinEntry::String(script) => {
+          let name = &package.id.nv.name;
+          symlink_bin_entry(
+            name,
+            script,
+            &package_path,
+            &bin_node_modules_dir_path,
+          )?;
+        }
+        deno_npm::registry::NpmPackageVersionBinEntry::Map(entries) => {
+          for (name, script) in entries {
+            symlink_bin_entry(
+              name,
+              script,
+              &package_path,
+              &bin_node_modules_dir_path,
+            )?;
+          }
+        }
+      }
+    }
+  }
+
+  // 7. Run pre/post/install scripts for packages
+  for (package, package_path) in &*packages_with_install_scripts.lock() {
+    let package = snapshot.package_from_id(&package.id).unwrap();
+
+    for (script_name, script) in &package.scripts {
+      if script_name == "preinstall" || script_name == "install" || script_name == "postinstall" {
+        log::warn!(
+          "⚠️  {} {} has a \"{}\" script that might be required to execute for the package to work correctly.", 
+          deno_terminal::colors::yellow("Warning"), 
+          deno_terminal::colors::green(format!("{}", package.id.nv.name)), 
+          deno_terminal::colors::gray(format!("{}", script_name)),
+        );
+        log::warn!("  Script: {}", deno_terminal::colors::gray(script));
+        // TODO: add a prompt here to ask if we should run the script.
+        log::warn!("  Do you want to run this script with full permissions? [y/N]");
+        // ASCII code for the bell character.
+        print!("\x07");
+        eprint!(" > ");
```

Comment:
Could a previously executing script stuff stdin? We should maybe re-use the permission prompt code here somehow?